api/src/organizational-entities/infrastructure/repositories/organization-for-admin.repository.js

```diff
@@ -36,7 +36,7 @@ const archive = async function ({ id, archivedBy }) {
 
   await knex('memberships').where({ organizationId: id, disabledAt: null }).update({ disabledAt: archiveDate });
 
-  await knex(ORGANIZATIONS_TABLE_NAME)
+  return knex(ORGANIZATIONS_TABLE_NAME)
```

Comment:
Transformer la signature de cette fonction pour des facilités de tests ne me semble pas satisfaisant. Cela laisse penser qu'il y a un intérêt à récupérer la valeur de sortie de la fonction, alors qu'il n'y a pas de valeur ajoutée à le faire.

Par contre idéalement au niveau des tests d'intégration de ce repository il serait utile de tester que la fonction `archive` archive une et une seule organisation et pas d'autres, comme cela est testé pour les centres de certification dans `api/tests/organizational-entities/integration/infrastructure/repositories/certification-center-for-admin.repository.test.js`.

Reply:
Vu ensemble, le repo va maintenant throw en cas d'organisation not found